Add Ice Tea I2C communication test

maciejbocianski opened the pull request:
### Description

Provide I2C communication test which uses IceTea Framework.
Detailed description can be found in README.md file in the test case folder.

This PR depends on #8682 

### Pull request type

<!-- 
    Required
    Please add only one X to one of the following types. Do not fill multiple types (split the pull request otherwise).
    Please note this is not a GitHub task list, indenting the boxes or changing the format to add a '.' or '*' in front
    of them would change the meaning incorrectly. The only changes to be made are to add a description text under the
    description heading and to add a 'x' to the correct box.
-->
    [ ] Fix
    [ ] Refactor
    [ ] Target update
    [ ] Functionality change
    [ ] Docs update
    [X] Test update
    [ ] Breaking change

### Reviewers

@scartmell-arm 
@ARMmbed/mbed-os-hal

maciejbocianski committed:
Add Ice Tea I2C communication test

ciarmcom commented:
@maciejbocianski, thank you for your changes.
@scartmell-arm @ARMmbed/mbed-os-maintainers please review.

0xc0170 commented:
@scartmell-arm Please review

maciejbocianski committed:
i2c communication test update

 - sync with HAL I2C API changes
 - add stop signal configuration
 - add timeout functionality
 - update testcases

maciejbocianski committed:
i2c communication test: heap memory usage optimization

use stack memory for data buffers
